src/postamble.js

```diff
@@ -377,6 +377,24 @@ function exit(status, implicit) {
     return;
   }
 
+#if USE_PTHREADS
+  if (!implicit) {
+    if (ENVIRONMENT_IS_PTHREAD) {
+#if ASSERTIONS
+      err('Pthread 0x' + _pthread_self().toString(16) + ' called exit(), posting exitProcess.');
+#endif
+      // When running in a pthread we propagate the exit back to the main thread
+      // where it can decide if the whole process should be shut down or not.
```

Comment:
```suggestion
      // where it can decide if the whole process should be shut down or not.
      // The pthread may have decided not to exit its runtime, for example
      // because it runs a main loop, but that doesn't affect the main thread.
```